packages/vinxi/lib/router-modes.js

```diff
@@ -368,7 +368,7 @@ const routerModes = {
 		resolveConfig(router, appConfig, order) {
 			invariant(router.mode === "spa", "Invalid router mode");
 			const appRoot = appConfig.root ?? process.cwd();
-			const root = router.root ?? appRoot;
+			const root = resolve.absolute(router.root, appRoot) ?? appRoot;
```

Comment:
Requires some testing. It should also work with only `resolve.absolute(router.root, appRoot)`.